File: gemini/nivel basico/javascript/suma.js
function sumarNumeros(a, b) {

    return a + b;
  }

  try {
    const num1Str = prompt("Ingresa el primer número:");
    const num2Str = prompt("Ingresa el segundo número:");
  
    if (num1Str === null || num2Str === null) {
      console.log("Operación cancelada por el usuario.");
    } else {
      const num1 = parseFloat(num1Str);
      const num2 = parseFloat(num2Str);
  
      if (isNaN(num1) || isNaN(num2)) {
        console.error("Error: Por favor, ingresa números válidos.");
      } else {
        const resultado = sumarNumeros(num1, num2);
        console.log(`La suma de ${num1} y ${num2} es: ${resultado}`);
      }
    }
  } catch (e) {
    console.error("Ocurrió un error al procesar la entrada:", e);
  }
  

File: gemini/nivel basico/ruby/suma.rb
def sumar_numeros(a, b)
  a + b
end

puts "Ingresa el primer número:"
num1_str = gets.chomp
puts "Ingresa el segundo número:"
num2_str = gets.chomp

begin
  num1 = Float(num1_str)
  num2 = Float(num2_str)
  resultado = sumar_numeros(num1, num2)
  puts "La suma de #{num1} y #{num2} es: #{resultado}"
rescue ArgumentError
  puts "Error: Por favor, ingresa números válidos."
end